### DISCUSSION TITLE
All traffic comes from server location

### DISCUSSION THREAD

**Discussion by benjibenoit** [plausible/analytics#4353] [Self-Hosted Support]
Hi guys,
My server is hosted in germany with all my websites. When I look at my dashboard, it appears that all request are made from germany,
![image](https://github.com/user-attachments/assets/ec23789f-6e36-46a3-b774-218770a68632)

Is there a special config to add in this situation ?

**Comment by ruslandoga** [comment]
👋 @shankiflang 

Might be a reverse proxy / missing x-forwarded-for header issue. Please describe your setup in more detail.

**Reply by ruslandoga** [reply]
You can check what headers Plausible receives using tracing: https://github.com/plausible/analytics/discussions/2767#discussioncomment-5367417

**Reply by benjibenoit** [reply]
It might be possible, i'm actualy running it on Coolify with auto configuration

The docker compose file look like this :
```compose
services:
  plausible:
    image: 'plausible/analytics:v2.0'
    command: 'sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"'
    environment:
      - 'DATABASE_URL=postgres://postgres:$SERVICE_PASSWORD_POSTGRES@plausible_db/plausible'
      - BASE_URL=$SERVICE_FQDN_PLAUSIBLE_8000
      - SECRET_KEY_BASE=$SERVICE_BASE64_64_PLAUSIBLE
      - TOTP_VAULT_KEY=$SERVICE_REALBASE64_TOTP
    depends_on:
      - plausible_db
      - plausible_events_db
      - mail
    networks:
      ks4skoc: null
    labels:
      - coolify.managed=true
      - coolify.version=4.0.0-beta.294
      - coolify.serviceId=5
      - coolify.type=service
      - coolify.name=plausible-ks4skoc
      - coolify.pullRequestId=0
      - coolify.service.subId=10
      - coolify.service.subType=application
      - traefik.enable=true
      - traefik.http.middlewares.gzip.compress=true
      - traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https
      - traefik.http.routers.http-0-ks4skoc-plausible.entryPoints=http
      - traefik.http.routers.http-0-ks4skoc-plausible.middlewares=redirect-to-https
      - 'traefik.http.routers.http-0-ks4skoc-plausible.rule=Host(`XXX.com`) && PathPrefix(`/`)'
      - traefik.http.routers.https-0-ks4skoc-plausible.entryPoints=https
      - traefik.http.routers.https-0-ks4skoc-plausible.middlewares=gzip
      - 'traefik.http.routers.https-0-ks4skoc-plausible.rule=Host(`XXX.com`) && PathPrefix(`/`)'
      - traefik.http.routers.https-0-ks4skoc-plausible.tls.certresolver=letsencrypt
      - traefik.http.routers.https-0-ks4skoc-plausible.tls=true
      - 'caddy_0.encode=zstd gzip'
      - 'caddy_0.handle_path.0_reverse_proxy={{upstreams}}'
      - 'caddy_0.handle_path=/*'
      - caddy_0.header=-Server
      - 'caddy_0.try_files={path} /index.html /index.php'
      - 'caddy_0=https://XXX.com'
      - caddy_ingress_network=ks4skoc
    restart: unless-stopped
    container_name: plausible-ks4skoc
  mail:
    image: bytemark/smtp
    networks:
      ks4skoc: null
    labels:
      - coolify.managed=true
      - coolify.version=4.0.0-beta.294
      - coolify.serviceId=5
      - coolify.type=service
      - coolify.name=mail-ks4skoc
      - coolify.pullRequestId=0
      - coolify.service.subId=11
      - coolify.service.subType=application
    restart: unless-stopped
    container_name: mail-ks4skoc
  plausible_db:
    image: 'postgres:14-alpine'
    volumes:
      - 'ks4skoc_db-data:/var/lib/postgresql/data'
    environment:
      - POSTGRES_DB=plausible
      - POSTGRES_PASSWORD=$SERVICE_PASSWORD_POSTGRES
    networks:
      ks4skoc: null
    labels:
      - coolify.managed=true
      - coolify.version=4.0.0-beta.294
      - coolify.serviceId=5
      - coolify.type=service
      - coolify.name=plausible_db-ks4skoc
      - coolify.pullRequestId=0
      - coolify.service.subId=9
      - coolify.service.subType=database
    restart: unless-stopped
    container_name: plausible_db-ks4skoc
  plausible_events_db:
    image: 'clickhouse/clickhouse-server:23.3.7.5-alpine'
    volumes:
      -
        type: volume
        source: ks4skoc_event-data
        target: /var/lib/clickhouse
      -
        type: bind
        source: ./clickhouse/clickhouse-config.xml
        target: /etc/clickhouse-server/config.d/logging.xml
        read_only: true
      -
        type: bind
        source: ./clickhouse/clickhouse-user-config.xml
        target: /etc/clickhouse-server/users.d/logging.xml
        read_only: true
    ulimits:
      nofile:
        soft: 262144
        hard: 262144
    networks:
      ks4skoc: null
    labels:
      - coolify.managed=true
      - coolify.version=4.0.0-beta.294
      - coolify.serviceId=5
      - coolify.type=service
      - coolify.name=plausible_events_db-ks4skoc
      - coolify.pullRequestId=0
      - coolify.service.subId=10
      - coolify.service.subType=database
    restart: unless-stopped
    container_name: plausible_events_db-ks4skoc
volumes:
  ks4skoc_db-data:
    name: ks4skoc_db-data
  ks4skoc_event-data:
    name: ks4skoc_event-data
networks:
  ks4skoc:
    name: ks4skoc
    external: true

```

**Reply by ruslandoga** [reply]
Another related discussion: https://github.com/plausible/analytics/discussions/3248

**Reply by ruslandoga** [reply]
You seem to be using both Caddy and Traefik for proxying the requests to Plausible. That might be problematic. One of them might be dropping the relevant (e.g x-forwarded-for) headers.

**Reply by benjibenoit** [reply]
> You seem to be using both Caddy and Traefik for proxying the requests to Plausible. That might be problematic. One of them might be dropping the relevant (e.g x-forwarded-for) headers.

Is there a config I can try to see which on is blocking ?

**Reply by ruslandoga** [reply]
I have never used Coolify nor Traefik, so I wouldn't know, sorry. There is a way to trace the headers that Plausible receives, though: https://github.com/plausible/analytics/discussions/4353#discussioncomment-10049255

**Reply by ruslandoga** [reply]
You can also try finding access logs for both Caddy and Traefik and figure out where the headers are dropped (if they are dropped, please check Plausible first) from that.

**Reply by benjibenoit** [reply]
> You can also try finding access logs for both Caddy and Traefik and figure out where the headers are dropped (if they are dropped, please check Plausible first) from that.

I see in both docs that it auto passes the headers. I'm trying to connect to docker with your reply but I get en empty docker, dono why.

**Reply by ruslandoga** [reply]
What do you mean by "empty docker"? What exact steps are you performing?

**Reply by benjibenoit** [reply]
> What do you mean by "empty docker"? What exact steps are you performing?

I just `docker ps` on my machine hosting coolify and there is no container while I've got my whole production going. I'll check when asap

**Reply by benjibenoit** [reply]
I've update my cloudflare to disable proxy and it works. https://github.com/plausible/analytics/discussions/2767#discussioncomment-5364387 I need this proxy for my prod protection. Do you know if there is any alternative to keep the cf proxy ?

**Reply by ruslandoga** [reply]
It works by default with Plausible as explained in that comment.

**Reply by benjibenoit** [reply]
> You can also try finding access logs for both Caddy and Traefik and figure out where the headers are dropped (if they are dropped, please check Plausible first) from that.

I removed the proxy of plausible on cloudflare and it seems to work now. How it worked then ?

**Reply by ruslandoga** [reply]
How what worked when?

**Reply by benjibenoit** [reply]
> How what worked when?

The geoloc work when I disable the proxy on plausible domain on cloudflare
